### ISSUE TITLE
Constraint violation during warm boot

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#500]
@vvidic Observed the following on Debian:

```
Jul  5 07:13:38 autopkgtest-lxc-gnzvcc safe_run/ticker ERROR    Exception: IntegrityError: duplicate key value violates unique constraint "test_name_key"
DETAIL:  Key (name)=(trace) already exists.
CONTEXT:  SQL statement "INSERT INTO test (json, updated, available)
      VALUES (new_json, now(), true)"
PL/pgSQL function test_upsert(jsonb) line 14 at SQL statement
SQL statement "SELECT test_upsert(test_enumeration)"
PL/pgSQL function test_boot() line 43 at PERFORM
SQL statement "SELECT test_boot()"
PL/pgSQL function warm_boot() line 4 at PERFORM
SQL statement "SELECT warm_boot()"
PL/pgSQL function cold_boot() line 3 at PERFORM

Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pscheduler/saferun.py", line 41, in safe_run
    function()
  File "/usr/lib/pscheduler/daemons/ticker", line 156, in <lambda>
    pscheduler.safe_run(lambda: main_program())
  File "/usr/lib/pscheduler/daemons/ticker", line 127, in main_program
    cursor.execute("SELECT cold_boot()")

IntegrityError: duplicate key value violates unique constraint "test_name_key"
DETAIL:  Key (name)=(trace) already exists.
CONTEXT:  SQL statement "INSERT INTO test (json, updated, available)
      VALUES (new_json, now(), true)"
PL/pgSQL function test_upsert(jsonb) line 14 at SQL statement
SQL statement "SELECT test_upsert(test_enumeration)"
PL/pgSQL function test_boot() line 43 at PERFORM
SQL statement "SELECT test_boot()"
PL/pgSQL function warm_boot() line 4 at PERFORM
SQL statement "SELECT warm_boot()"
PL/pgSQL function cold_boot() line 3 at PERFORM
```
Valentin's theory (probably correct) is that this is happening during an install or an upgrade where the ticker is starting up and doing a cold boot while other packages might be doing the same.

In theory, this shouldn't matter because it's all supposed to be wrapped in transactions.  Need to see if this can be reproduced and what the cause is.

**Comment by mfeit-internet2** [comment]
This doesn't seem to be a problem and appears to have been fixed in Debian by 375b7ab411dca3d43c7c42c5f19e84c4472a411d.

@vvidic If you're good with this on Debian, please close this ticket.

**Comment by vvidic** [comment]
Yep, can't seem to reproduce it now. Will let you know if it appears again.
